include/faabric/scheduler/FunctionMigrationThread.h

```diff
@@ -1,26 +1,23 @@
 #pragma once
 
+#include <faabric/util/PeriodicBackgroundThread.h>
+
 #include <condition_variable>
 #include <mutex>
 #include <thread>
 
+using namespace faabric::util;
+
 namespace faabric::scheduler {
-// Start a background thread that, every wake up period, will check if there
-// are migration opportunities for in-flight apps that have opted in to
-// being checked for migrations.
-class FunctionMigrationThread
+
+/**
+ * Background thread that, every wake up period, will check if there
+ * are migration opportunities for in-flight apps that have opted in to
+ * being checked for migrations.
+ */
+class FunctionMigrationThread : public PeriodicBackgroundThread
```

Comment:
I've refactored this to extract the periodic-wakeup-thread bits into `PeriodicBackgroundThread`.